src/validation/custom.validation.ts

```diff
@@ -93,14 +93,23 @@ export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", ye
 };
 
 export const checkIdentityDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
-  const isMandatoryDateValid = checkMandatoryDate(dayStr, monthStr, yearStr);
-  if (isMandatoryDateValid) {
-    checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
+  const isDatePresent = checkDateIsNotCompletelyEmpty(ErrorMessages.ENTER_DATE, dayStr, monthStr, yearStr);
```

Comment:
the checks in here look the same as those in checkStartDate plus the extra last 3 months check. You could rename checkStartDate to something like checkDateIsValidAndInPast and call that from within checkIdentityDate.
Or if you want to keep the checkStartDate name, you could create a new function like checkDateIsValidAndInPast and have checkStartDate just call checkDateIsValidAndInPast, then have checkIdentityDate call checkDateIsValidAndInPast and checkDateIsWithinLast3Months

Reply:
Trying to reuse code like this will present a complication here because of checkAllDateFieldsArePresent returning true or false and so checkDateIsValidAndInPast would have to as well as we would require a check from the result of that. This may not be obvious from just looking at the code, you may need to check the branch out and try this proposal to see what I mean here.

Reply:
i see now, just tried it out locally, if the areAllDateFieldsPresent returns false then the last line of the method we are re-using will return true instead. ok, ignore this comment.